lib/tree.rb

```diff
@@ -30,6 +30,14 @@ def contents_array
       @contents_ary ||= jtree_entries
     end
     
+    def recursive_count(limit = nil)
+      jtree_entries({recursive: true, limit: limit}).size
```

Comment:
Yes, that's what I was thinking. Not sure if it's worth the trouble, but I suppose we could cache the limit as well, and rerun only if the limit changes?